examples/arp_response.rs

```diff
@@ -131,34 +140,77 @@ fn main() {
 }
 
 fn process_packets(
-    nic: &mut PvNic,
-    packets: &mut Vec<PvPacket>,
+    nic: &mut pv::Nic,
+    packets: &mut Vec<pv::Packet>,
     batch_size: u32,
     src_mac_address: &MacAddr,
 ) -> u32 {
     let mut processed = 0;
     for i in (0..batch_size).rev() {
-        let payload_ptr = unsafe {
-            packets[i as usize]
-                .buffer
-                .add(packets[i as usize].start as usize)
-                .cast_const()
-        };
-
-        // analyze packet and create packet
-        unsafe {
-            if std::ptr::read(payload_ptr.offset(12)) == 0x08
-                && std::ptr::read(payload_ptr.offset(13)) == 0x06
-                && std::ptr::read(payload_ptr.offset(20)) == 0x00
-                && std::ptr::read(payload_ptr.offset(21)) == 0x01
-            // ARP request packet
-            {
-                gen_arp_response_packet(&mut packets[i as usize], src_mac_address);
-                processed += 1;
-            } else {
-                pv_free(nic, packets, i as usize);
+        // analyze packet
+        let packet_kind: PacketKind = is_arp_req(&packets[i as usize]);
+
+        // process packet
+        match packet_kind {
+            PacketKind::ArpReq => {
+                let processing_result: Result<u32, u32> =
+                    make_arp_response_packet(src_mac_address, &mut packets[i as usize]);
+
+                if processing_result.is_ok() {
```

Comment:
```suggestion
                if make_arp_response_packet(src_mac_address, &mut packets[i as usize]).is_ok() {
```

Reply:
the previous changed code was pushed later this suggestion, I committed this [1108e52](https://github.com/tsnlab/packetvisor/pull/47/commits/1108e52c05213069baaa773a3e943e011dbd8b06) using `--author` option to tag you